File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mino = prices.front();
        int maxprofit = 0;

        for(int i = 0; i < prices.size(); i++)
        {
            int profit = prices[i] - mino;

            maxprofit = max(maxprofit,profit);

            if(mino > prices[i])
            {
                mino = prices[i];
            }
           
        }

        return maxprofit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int n = prices.length;
        int first = prices[0];

        for(int i = 1; i < n; i++)
        {
            int profit = prices[i] - first;
            maxProfit = Math.max(maxProfit,profit);
            
            first = Math.min(first,prices[i]);
        }

        return maxProfit;
    }
}